### PR TITLE
fix: Update ps2census to 0.2.4

### LINKED ISSUE #36: Update ps2census to latest version (0.2.4)
The following things have to be fixed, when we update to the latest ps2census version (current version is 0.1.2):

- [x] CensusProxy.ts --> enable 'ContinentUnlock'
- [x] ContinentUnlockEvent.ts --> Remove event as unknown cast
- [x] Swap from GenericEvent to PS2Event

### PR DESCRIPTION
**Pull request by marci4**
Fixes #36 

The following issues were solved:
- [x] CensusProxy.ts --> enable 'ContinentUnlock'
- [x] ContinentUnlockEvent.ts --> Remove event as unknown cast
- [X] Swap from GenericEvent to PS2Event

### CODE REVIEW

File: app/src/handlers/census/CensusProxy.ts [hunk 1]
```diff
@@ -23,6 +28,11 @@ export default class CensusProxy {
     private readonly continentLockHandler: ContinentLockEventHandler;
     private readonly facilityControlEventHandler: FacilityControlEventHandler;
     private readonly gainExperienceEventHandler: GainExperienceEventHandler;
+    private readonly achievementEarnedHandler: AchievementEarnedHandler;
```
Comment (Maelstromeous) [inline app/src/handlers/census/CensusProxy.ts:31]: These were missing? 😲

File: app/src/handlers/census/CensusProxy.ts [hunk 2]
```diff
@@ -98,10 +117,9 @@ export default class CensusProxy {
             case 'ContinentLock':
                 this.continentLockHandler.handle(event);
                 break;
-            // Documented but never happens
-            // case 'ContinentUnlock':
-            //     eventStore.storeContinentUnlock(payload);
-            //     break;
+            case 'ContinentUnlock':
```
Comment (Maelstromeous) [inline app/src/handlers/census/CensusProxy.ts:120]: Still doesn't happen, how come you uncommented?

File: app/src/handlers/census/events/AchievementEarnedEvent.ts [hunk 3]
```diff
@@ -25,7 +24,7 @@ export default class AchievementEarnedEvent {
     public readonly timestamp: number;
 
     constructor(
-        event: GenericEvent,
+        event: PS2Event,
     ) {
         const achievementEarnedEvent = event as AchievementEarned;
```
Comment (Maelstromeous) [inline app/src/handlers/census/events/AchievementEarnedEvent.ts:29]: This appears to be a different format to the rest?

File: app/src/handlers/census/PlayerLogoutEventHandler.ts [hunk 4]
```diff
@@ -1,12 +1,12 @@
 import {inject, injectable} from 'inversify';
 import EventHandlerInterface from '../../interfaces/EventHandlerInterface';
-import {GenericEvent} from 'ps2census/dist/client/utils/PS2Events';
 import {getLogger} from '../../logger';
 import config from '../../config';
 import {jsonLogOutput} from '../../utils/json';
 import PlayerLogoutEvent from './events/PlayerLogoutEvent';
 import PlayerHandlerInterface from '../../interfaces/PlayerHandlerInterface';
 import {TYPES} from '../../constants/types';
+import {PS2Event} from 'ps2census';
```
Comment (Maelstromeous) [inline app/src/handlers/census/PlayerLogoutEventHandler.ts:9]: Other places you've imported the specific event type but not here?

**Comment by marci4** [comment]
Without any instanceof checks, we have no guarantee at runtime that the PS2Event is really a DeathEvent.

Would also make the HandlerInterface obsolete.

File: app/src/services/census/index.ts [hunk 5]
```diff
@@ -1,21 +1,21 @@
 import {ContainerModule} from 'inversify';
 import ServiceInterface, {SERVICE} from '../../interfaces/ServiceInterface';
 import config from '../../config';
-import PS2EventClient from 'ps2census/dist/client/Client'; // TODO: Await microwave's type fixes
 import CensusStreamService from './CensusStreamService';
-import {PS2ClientConfig} from 'ps2census/dist/client/utils/Types';
+import Client from 'ps2census/dist/client/Client';
+import {EventStreamManagerConfig} from 'ps2census/dist/client/utils/Types';
```
Comment (Maelstromeous) [inline app/src/services/census/index.ts:7]: @microwavekonijn isn't the client exposed from the root of your package?

**Comment by Maelstromeous** [comment]
> Without any instanceof checks, we have no guarantee at runtime that the PS2Event is really a DeathEvent.
> 
> Would also make the HandlerInterface obsolete.

Maybe we should add them then. I want to prevent the handlers recieving the wrong event. Perhaps throwing a `InvalidArgumentException`

File: app/src/handlers/census/CensusProxy.ts (hunk 2 above)
Comment (marci4) [reply app/src/handlers/census/CensusProxy.ts:120]: The problem was that ContinentUnlock was not part of GenericEvent.
Therefore I implemented the handler but was not able to call it.

**Comment by marci4** [comment]
> 
> 
> > Without any instanceof checks, we have no guarantee at runtime that the PS2Event is really a DeathEvent.
> > Would also make the HandlerInterface obsolete.
> 
> Maybe we should add them then. I want to prevent the handlers recieving the wrong event. Perhaps throwing a `InvalidArgumentException`

You will get an InvalidArgumentException most certainly anyway since all the constructors do undefined checks

**Comment by microwavekonijn** [comment]
You can use the individual events like facilityControl or itemAdded, those will have the specific event objects.

File: app/src/handlers/census/CensusProxy.ts (hunk 1 above)
Comment (marci4) [reply app/src/handlers/census/CensusProxy.ts:31]: Yep, not sure what happened. Will include some tests later to catch such errors

File: app/src/services/census/index.ts (hunk 5 above)
Comment (marci4) [reply app/src/services/census/index.ts:7]: My bad, client is available but EventStreamManagerConfig is not exported

Comment (Maelstromeous) [reply app/src/services/census/index.ts:7]: @microwavekonijn making you aware certain types aren't exposed

Comment (microwavekonijn) [reply app/src/services/census/index.ts:7]: Oh, shoot forgot to export that file completely.

**Pull request merged**